### ISSUE TITLE
Long hyperlinks prevent workbook from loading on Windows

### ISSUE THREAD

**Issue by olivroy** [JanMarvin/openxlsx2#753]
![image](https://github.com/JanMarvin/openxlsx2/assets/52606734/6efd1b7a-281c-4050-9a92-91707e68fd00)

[test-long-hl.xlsx](https://github.com/JanMarvin/openxlsx2/files/12412663/test-long-hl.xlsx)

```r
wb_load("test-long-hl.xlsx")
#> Error in basename(xml_relship$Target) : path too long
```

![image](https://github.com/JanMarvin/openxlsx2/assets/52606734/f1112d36-479a-4b56-bc26-6c4edcbc8989)
```r
# To circumvent having long URL in a path or hyperlink
basename_long <- function(x) {
  # I am seeing this for paths longer than 260 on Window
s
  if (any(nchar(x) > 250)) {
    x <- stringi::stri_sub(x, from = -255L) # Cut the begining of the path
  }

  basename(x)
}
```

```r
# The link is the following
a_long_hl_from_google_search <- "https://www.google.com/search?q=a+very+long+hyperlink+from+a+google+search+aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa&sca_esv=559161978&sxsrf=AB5stBh3R0RwFy8QBN-mAmP3mYzwqhsCqw%3A1692733190678&ei=Bg_lZKz1KJ6I0PEPuYq-8A8&ved=0ahUKEwiskJ7EgvGAAxUeBDQIHTmFD_4Q4dUDCA8&uact=5&oq=a+very+long+hyperlink+from+a+google+search+aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa&gs_lp=Egxnd3Mtd2l6LXNlcnAiVGEgdmVyeSBsb25nIGh5cGVybGluayBmcm9tIGEgZ29vZ2xlIHNlYXJjaCBhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYTIFECEYoAEyBRAhGKABMgUQIRigATIFECEYoAFI2RFQkQFYxBBwAXgBkAEAmAG-AqAByAyqAQUyLTUuMbgBA8gBAPgBAcICChAAGEcY1gQYsAPiAwQYACBBiAYBkAYC&sclient=gws-wiz-serp&safe=active&ssui=on"
```

**Comment by JanMarvin** [comment]
Ah I see, hm, it's possibly a windows only restriction. R-core was messing with path length in Windows not to long ago. We don't have to use Basename() it was simply a convenient function, but we can use whatever function to get the base name. A small regex should do the trick.

**Comment by JanMarvin** [comment]
Loads fine on Mac. We can use `sub(".*/([^/]+)$", "\\1", path)` instead. It's way slower, but we have to use something

**Comment by olivroy** [comment]
What about?

```r
# only go there if error?
basename2 <- function(x) {
 tryCatch(basename(x), error = function(x) sub(".*/([^/]+)$", "\\1", x)
}
```

**Comment by JanMarvin** [comment]
Nah, I'll think of something.

**Comment by olivroy** [comment]
Closed by #755
